litmus/director/openebs-self-install/test.yml

```diff
@@ -0,0 +1,201 @@
+        - name: Fetch the project id of the project where the cluster is running
+          uri:
+            url: "{{ director_url }}/v3/groups/{{ group_id }}/project"
+            method: GET
+            url_username: "{{ username.stdout }}"
+            url_password: "{{ password.stdout }}"
+            force_basic_auth: yes
+            return_content: yes
+            status_code: 200
+          register: project_details
+
+        ## Saving Project id
+        - name: Saving the project id
+          set_fact:
+            project_id: "{{ project_details.json.data[0].id }}"
+
+        ## Getting the total number of nodes in the cluster
+        - name: Getting the number of nodes in the cluster
+          shell: kubectl get nodes --no-headers | wc -l
+          args:
+            executable: /bin/bash
+          register: nodes
+
+        ## Fail the test if it contains less than two nodes
+        ## Minimum two nodes required
+        - name: Checking if the cluster contains atleast two nodes
+          fail:
+            msg: Minimum two nodes is required to install openebs in this cluster
+          when: "{{ nodes.stdout }} < 2"
+
+        ## Getting node details of the cluster
+        ## Selecting the node which is in active state present in the cluster
+        - name: Getting the node details of the cluster
+          uri:
+            url: "{{ director_url }}/v3/groups/{{ group_id }}/nodes?state=active&clusterId={{ cluster_id }}"
+            method: GET
+            url_username: "{{ username.stdout }}"
+            url_password: "{{ password.stdout }}"
+            force_basic_auth: yes
+            return_content: yes
+            body_format: json
+            body:
+              status_code: 202
+          register: node_details
+
+        ## Define variable node_id
+        - set_fact:
+            node_id: []
+
+        ## Storing the node id of the nodes in the cluster
+        - name: Storing the node id of node in the cluster
+          set_fact:
+            node_id: "{{ node_id  + [item.id] }}"
+          loop: "{{ node_details.json.data }}"
+
+
+        ## Labeling the node-1 of the Cluster with controlPlaneNode=true and dataPlaneNode=true
+        ## With first element in the node_id list
+        - name: Labeling the node-1 of the cluster
+          uri:
+            url: "{{ director_url }}/v3/groups/{{ group_id }}/nodes/{{ node_id[0] }}/?action=labelnodes"
+            method: POST
+            url_username: "{{ username.stdout }}"
+            url_password: "{{ password.stdout }}"
+            force_basic_auth: yes
+            return_content: yes
+            body_format: json
+            body: '{"controlPlaneNode": true, "dataPlaneNode": true}'
+            status_code: 202
+          register: labelnode1
+
+        ## Labeling node-2 of the Cluster with controlPlaneNode=true and dataPlaneNode=false
+        ## With the second element in the node_id list
+        - name: labeling node-2 of the cluster
```

Comment:
Start the task name with a capital letter. This is just to ensure that we follow a similar convention all across the tasks.